### PR TITLE
Fix inconsistent interpolation evaluation due to Julia bug

### PR DESCRIPTION
**Pull request by mortenpi**
I noticed that the tests that parse this string

```julia
value = 1
ast = cm"$(value) $(value + 1) $(value += 1) $(value += 1)"
```

intermittently fail. It turns out that assignments in argument lists (which `cm""` depends on, when it constructs the vector of values) are problematic: https://github.com/JuliaLang/julia/issues/46251

This changes the `cm""` macro such that the generated code evaluates the values one-by-one and pushes them into a vector that later gets passed to `_interp!`, rather than constructing the vector in one go with a `[ ... ]` expression. This should make sure that values are evaluated in a deterministic order.

I split the commits to see if the MWE test also fails on CI systematically, but I think the PR should be squash-merged.

### CODE REVIEW

File: test/extensions/interpolation.jl [hunk 1]
```diff
@@ -79,4 +89,15 @@ end
...
+    # Make sure that the evaluation of values happens at runtime.
+    f(x) = cm"if x = $(x), then x² = $(x^2)"
+    let ast = f(2)
+        @test markdown(ast) == "if x = \$(x), then x² = \$(x ^ 2)\n"
+        @test term(ast) == " if x = \e[33m2\e[39m, then x² = \e[33m4\e[39m\n"
+    end
+    let ast = f(-3)
+        @test markdown(ast) == "if x = \$(x), then x² = \$(x ^ 2)\n"
+        @test term(ast) == " if x = \e[33m-3\e[39m, then x² = \e[33m9\e[39m\n"
+    end
```
Comment (mortenpi) [inline test/extensions/interpolation.jl:102]: These tests are unrelated, but cover a small edge case, where theoretically it might be possible to have `cm` evaluate at parse time.

**Review by MichaelHatherly — APPROVED** [review]
Cool bug! Thanks for the fixes.

**Comment by mortenpi** [comment]
I just discovered that the inconsistent evaluation is probably consistent with normal string interpolation:

```
julia> value = 1; "$(value), $(value + 1), $(value += 1), $(value += 1)"
"3, 2, 2, 3"

julia> let value = 1; "$(value), $(value + 1), $(value += 1), $(value += 1)"; end
"1, 2, 2, 3"
```

So I am not sure if CommonMark should fix this for `cm""`, or stay consistent with normal string interpolation.

**Comment by MichaelHatherly** [comment]
Ooh, very good point... I reckon best to keep the "fixed" behaviour you've added here rather than matching a bug upstream, which, hopefully, will get fixed at some point.
